### PR TITLE
ClientError rescuable from Base module to all resources

### LINKED ISSUE #15: Translate Faraday::ClientError into IEX::Errors::ClientError everywhere
We now handle Faraday::ClientError differently in Chart#get than in all the other calls. Error handling should be moved into the base class and handled consistently everywhere.

@rodolfobandeira 

Add a module ala

```ruby
module IEX::Resources::Base
   def self.symbol(symbol, &_block)
      yield
     rescue Faraday::ResourceNotFound => e
        raise IEX::Errors::SymbolNotFoundError.new(symbol, e.response[:body])
      rescue Faraday::ClientError => e
        raise IEX::Errors::ClientError, e.response
      end
   end
```

Then `extend` other modules with it and use `symbol do ...` everywhere instead of explicitly rescue-ing exceptions.

### PR DESCRIPTION
**Pull request by rodolfobandeira**
close #15 

cc: @dblock

### CODE REVIEW

File: lib/iex/resources/chart.rb [hunk 1]
```diff
@@ -1,28 +1,27 @@
+require_relative 'base'
 require_relative 'chart/base'
```
Comment (rodolfobandeira) [inline lib/iex/resources/chart.rb]: What's your thoughts about the lines above? It is clear that they are distinct base classes but I don't know about having the same name. If you see bellow, I'm calling Base.symbol passing the full path so there is no conflict. Should we keep it?

File: lib/iex/resources/company.rb [hunk 2]
```diff
@@ -11,10 +13,10 @@ class Company < Resource
       property 'issue_type', from: 'issueType' # common issue type of the stock
       property 'sector'
 
-      def self.get(symbol)
-        new IEX::Api::Company.get(symbol)
-      rescue Faraday::ResourceNotFound => e
-        raise IEX::Errors::SymbolNotFoundError.new(symbol, e.response[:body])
+      def self.get(stock_symbol)
```
Comment (rodolfobandeira) [inline lib/iex/resources/company.rb]: Decided to rename our parameter to `stock_symbol` instead to differentiate from  `Base.symbol` method name.

File: lib/iex/resources/chart.rb (hunk 1 above)
Comment (dblock) [reply lib/iex/resources/chart.rb]: It's totally fine.

**Comment by dblock** [comment]
Looks great, merging.

**Pull request merged**